scene/2d/physics_body_2d.cpp

```diff
@@ -1044,6 +1044,8 @@ void RigidBody2D::_reload_physics_characteristics() {
 #define FLOOR_ANGLE_THRESHOLD 0.01
 
 bool CharacterBody2D::move_and_slide() {
+	ERR_FAIL_COND_V(up_direction == Vector2(), false);
```

Comment:
This shouldn't happen since it's already checked in the setter.

Reply:
I did this for 2 reasons (but it maybe useless):

- If you import an old project, to avoid that case
- if we decide to remove the restriction on the setter because users find that too bothering to not be able to setup the up direction to Vector2.ZERO temporarily, we avoid issue.

What do you think?

Reply:
It's not really needed unless we decide to remove the restriction on the setter.
For info, when loading a project it still goes through the setter for each property so that shouldn't be a problem.

If we decide to go for no restriction in the setter, then it can be handled with an error in move like you did (but maybe just once and using a fallback instead of returning immediately).

Something like:
```
Vector2 used_up_direction = up_direction;
if (unlikely(up_direction == Vector2())) {
    ERR_PRINT_ONCE("up_direction can't be equal to Vector2.ZERO, using Vector2.UP instead. Consider using Free motion mode instead.");
    used_up_direction = Vector2(0.0, -1.0);
}
```

And then maybe a warning in the editor could also help.